sympy/printing/latex.py

```diff
@@ -1813,7 +1813,50 @@ def _print_PolynomialRingBase(self, expr):
 
     def _print_Poly(self, poly):
         cls = poly.__class__.__name__
-        expr = self._print(poly.as_expr())
+        terms, gens = [], [s for s in poly.gens]
+        for monom, coeff in poly.terms():
+            s_monom = None
```

Comment:
I think s_monom should start out being an empty string, which will be extended in the loop immediately below.